server.js

```diff
@@ -1,26 +1,36 @@
 const express = require('express');
 const bodyParser = require('body-parser');
 const app = express();
-
+// config contains values that are read in from the .env file
+// and hidden from the application, otherwise its a security risk
+const config = require("./config");
 const toDo = require('./routes/to-do');
-
 const mongoose = require('mongoose');
-let dev_db_url = 'mongodb://sxiong:siaxiong1@ds111113.mlab.com:11113/todolist';
-let mongoDB = process.env.MONGODB_URI || 'mongodb://sxiong:siaxiong1@ds111113.mlab.com:11113/todolist'
-mongoose.connect(mongoDB);
+
+let dev_db_url = config.production.mongoConnectionString;
+let mongoConnectionString = process.env.MONGODB_URI || config.production.MONGODB_URI
+
+mongoose.connect(mongoConnectionString, {
+  useNewUrlParser: true,
+  auth: {
+    user: config.production.USER,
+    password: config.production.PASS
+  }
+});
 mongoose.Promise = global.Promise;
+
 let db = mongoose.connection;
 db.on('error', console.error.bind(console, 'MongoDB connection error:'));
+db.once('open', function callback () {
```

Comment:
This will let you know that the db is connected and what string it used to connect to.